src/file_pipe_log/format.rs

```diff
@@ -155,6 +156,59 @@ impl LogFileFormat {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum Content {
+    Checksum(u32),
+}
+
+#[derive(Debug, Clone)]
+pub struct LogFileContext {
+    pub id: FileId,
+    pub version: Version,
+    _cont: Option<Content>,
```

Comment:
There is no need to keep a copy of content. `xor` is arguably faster than integer addition, there's no need to reduce its cost.